src/rust/engine/src/externs/interface.rs

```diff
@@ -1626,19 +1626,27 @@ fn write_digest(
   py_session: &PySession,
   digest: &PyAny,
   path_prefix: String,
+  clear_destination: Option<bool>,
 ) -> PyO3Result<()> {
   let core = &py_scheduler.0.core;
   core.executor.enter(|| {
     // TODO: A parent_id should be an explicit argument.
     py_session.0.workunit_store().init_thread_state(None);
 
     let lifted_digest = nodes::lift_directory_digest(digest).map_err(PyValueError::new_err)?;
+    let clear_destination = clear_destination.unwrap_or(false);
 
     // Python will have already validated that path_prefix is a relative path.
     let mut destination = PathBuf::new();
     destination.push(core.build_root.clone());
     destination.push(path_prefix);
 
+    if clear_destination {
+     let _result = std::fs::remove_dir_all(&destination).map_err(|e| {
+       PyException::new_err(format!("Failed to clear {}: {e}", destination.display()))
+     });
```

Comment:
To actual return an error if it occurs here, you should use `?` ([see](https://doc.rust-lang.org/rust-by-example/std/result/question_mark.html)):
```suggestion
     std::fs::remove_dir_all(&destination).map_err(|e| {
       PyException::new_err(format!("Failed to clear {}: {e}", destination.display()))
     })?;
```